compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1139,29 +1282,19 @@ impl Context {
     ) -> Result<AcirVar, RuntimeError> {
         let element_type_sizes = self.init_element_type_sizes_array(array_typ, array_id, dfg)?;
 
-        let element_size = array_typ.element_size();
+        let true_pred =
+            self.acir_context.mul_var(var_index, self.current_side_effects_enabled_var)?;
+        let one = self.acir_context.add_constant(FieldElement::one());
+        let not_pred = self.acir_context.sub_var(one, self.current_side_effects_enabled_var)?;
 
-        let element_size_var =
-            self.acir_context.add_constant(FieldElement::from(element_size as u128));
-        let outer_offset = self.acir_context.div_var(
-            var_index,
-            element_size_var,
-            AcirType::unsigned(32),
-            self.current_side_effects_enabled_var,
-        )?;
-        let inner_offset_index = self.acir_context.modulo_var(
-            var_index,
-            element_size_var,
-            32,
-            self.current_side_effects_enabled_var,
-        )?;
-        let inner_offset =
-            self.acir_context.read_from_memory(element_type_sizes, &inner_offset_index)?;
+        let zero = self.acir_context.add_constant(FieldElement::zero());
+        let false_pred = self.acir_context.mul_var(not_pred, zero)?;
+        let var_index = self.acir_context.add_var(true_pred, false_pred)?;
```

Comment:
The goal was that in the case of a true predicate we use the normal var index, while in the case of a false predicate we just set the var index to zero as otherwise we could have memory out of bounds errors. As I am just reading from zero in the false though, I see what you mean about them being equivalent. Just want to test it to make sure before committing

Reply:
Ah ok, that makes sense. We should be able to simplify this all down to `let pred_var_index =
            self.acir_context.mul_var(var_index, self.current_side_effects_enabled_var)?;` in that case.

Reply:
Yup you are correct, done.